lib/streaming.js

```diff
@@ -110,11 +119,17 @@ StreamProxy.prototype._onFinish = function() {
 StreamProxy.prototype.setStream = function(apiCall, argument) {
   var stream = apiCall(argument, this._callback);
   this.stream = stream;
-  stream.on('error', this.emit.bind(this, 'error'));
+  (['error', 'metadata', 'status']).forEach(function(event) {
+    stream.on(event, this.emit.bind(this, event));
+  }.bind(this));
+  // We also want to supply the status data as 'response'.
+  stream.on('status', this.emit.bind(this, 'response'));
```

Comment:
Updated the code here.

See the discussion from https://github.com/GoogleCloudPlatform/google-cloud-node/pull/1775#issuecomment-259141029 for the details, but in short, this is expected by google-cloud-node -- emits `response` object in a certain format upon `metadata` event (not `status` event which I wrote). I've also added more comments with links so things will be clearer now. PTAL.